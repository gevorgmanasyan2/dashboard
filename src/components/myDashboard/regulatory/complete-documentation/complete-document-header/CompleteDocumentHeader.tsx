import React, { FC, ReactElement, ReactNode } from 'react'
import { Tooltip } from 'antd'
interface CompleteDocumentHeaderProps {
  bgColor?: string
  date?: string | number
  icon?: ReactElement | {}
  value?: string
  tooltip?: string | ReactNode
  onClick?: (e: any) => void
  id?: string | number
}

const CompleteDocumentHeader: FC<CompleteDocumentHeaderProps> = ({
  bgColor,
  date,
  icon,
  value,
  tooltip,
  onClick,
  id,
}) => {
  return (
    <button value={id} className="doc__container" onClick={onClick}>
      <div style={{ background: bgColor }} className="doc__container__date">
        <span>
          <Tooltip title={tooltip}> {date}</Tooltip>
        </span>
      </div>
      <div style={{ color: bgColor }} className="doc__container__content">
        {icon}
        <span>{value}</span>
      </div>
    </button>
  )
}

export default CompleteDocumentHeader
