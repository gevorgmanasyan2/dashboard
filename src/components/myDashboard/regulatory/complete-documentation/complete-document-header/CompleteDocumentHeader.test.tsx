/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import CompleteDocumentHeader from './CompleteDocumentHeader'
import { CheckCircleFilled } from '@ant-design/icons'

it('should render CompleteDocumentHeader component', () => {
  const mockItem = {
    dateText: 'Completed Date',
    bgColor: '"#26A744"',
    date: '2020-07-10',
    icon: <CheckCircleFilled />,
    status: 'Initial Review',
  }
  expect(shallow(<CompleteDocumentHeader {...mockItem} />)).toMatchSnapshot()
})
