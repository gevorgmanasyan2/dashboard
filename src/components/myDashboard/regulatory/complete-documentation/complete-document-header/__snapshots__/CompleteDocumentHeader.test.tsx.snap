// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CompleteDocumentHeader component 1`] = `
<button
  className="doc__container"
>
  <div
    className="doc__container__date"
    style={
      Object {
        "background": "\\"#26A744\\"",
      }
    }
  >
    <span>
      Completed Date
      : 
      2020-07-10
    </span>
  </div>
  <div
    className="doc__container__content"
    style={
      Object {
        "color": "\\"#26A744\\"",
      }
    }
  >
    <ForwardRef(CheckCircleFilled) />
    <span />
  </div>
</button>
`;
